### PR TITLE
Switch to a `src` layout and pyproject.toml only

### PR DESCRIPTION
**Pull request by jhkennedy**
* `src` layout is recommended: https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout
* `setup.py` is depreciated/discouraged: https://setuptools.pypa.io/en/latest/userguide/quickstart.html#setuppy-discouraged

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -8,21 +8,21 @@ dependencies:
...
+  - setuptools_scm>=6.2
   - pytest
   - pytest-console-scripts
   - pytest-cov
   # For running
   - astropy
   - fiona
   - gdal>=3.3
-  - numpy
+  - numpy<1.24.0
```
Comment (jhkennedy) [inline environment.yml:25]: There are a lot of depreciations in numpy 1.24 and it seems to be messing with a lot of packages (also will unblock #154); we'll want to open a PR that unrestricts NumPy and periodically run the actions until it passes.

File: .github/workflows/static-analysis.yml [hunk 2]
```diff
@@ -20,31 +20,8 @@ jobs:
...
           flake8 --max-line-length=120 --import-order-style=pycharm --statistics \
-              --application-import-names asf_tools ArcGIS-toolbox/ASF_Tools.pyt asf_tools
+              --application-import-names asf_tools ArcGIS-toolbox/ASF_Tools.pyt src/asf_tools
 
 
   call-secrets-analysis-workflow:
-    uses: ASFHyP3/actions/.github/workflows/reusable-secrets-analysis.yml@v0.4.0
-
-
-  trufflehog:
```
Comment (forrestfwilliams) [inline .github/workflows/static-analysis.yml:30]: Are we removing our trufflehog static analysis, or has this been moved elsewhere?

**Review by forrestfwilliams — APPROVED** [review]
I'm not sure I understand all of the changes that have been made to the `distribute.yml` workflow, but everything looks good to me.

File: .github/workflows/distribute.yml [hunk 3]
```diff
@@ -6,50 +6,47 @@ on:
...
-          activate-environment: asf-tools
-          environment-file: environment.yml
+      - uses: mamba-org/provision-with-micromamba@v14
 
-      - name: Pip install ${{ needs.distribute.outputs.SDIST_VERSION }}
-        shell: bash -l {0}
+      - name: Ensure edlfs v${{ needs.call-version-info-workflow.outputs.version }}} is pip installable
         run: |
-          python -m pip install asf_tools==${{ needs.distribute.outputs.SDIST_VERSION }}
+          python -m pip install edlfs==${{ needs.call-version-info-workflow.outputs.version_tag }}
```
Comment (forrestfwilliams) [inline .github/workflows/distribute.yml:52]: Copy/paste issue here.

Comment (jhkennedy) [reply .github/workflows/distribute.yml]: WOOPS!

**Pull request merged**